server/enterprise/message_export/global_relay_export/global_relay_export.go

```diff
@@ -73,72 +67,39 @@ type Message struct {
 }
 
 func GlobalRelayExport(rctx request.CTX, p shared.ExportParams) (shared.RunExportResults, error) {
-	results := shared.RunExportResults{}
-	var attachmentsRemovedPostIDs []string
-	allExports := make(map[string][]*ChannelExport)
-
-	tmpFile, err := os.CreateTemp("", "")
+	exportData, err := shared.GetGenericExportData(p)
+	results := exportData.Results
 	if err != nil {
-		return results, fmt.Errorf("unable to open the temporary export file: %w", err)
-	}
-	defer file.DeleteTemp(rctx.Logger(), tmpFile)
-
-	zipFile := zip.NewWriter(tmpFile)
-
-	// postAuthorsByChannel is a map so that we don't store duplicate authors
-	postAuthorsByChannel := make(map[string]map[string]shared.ChannelMember)
-	metadata := shared.Metadata{
-		Channels:         p.ChannelMetadata,
-		MessagesCount:    0,
-		AttachmentsCount: 0,
-		StartTime:        p.BatchStartTime,
-		EndTime:          p.BatchEndTime,
+		return results, err
 	}
-	channelsInThisBatch := make(map[string]bool)
-
-	for _, post := range p.Posts {
-		channelId := *post.ChannelId
-		channelsInThisBatch[channelId] = true
 
-		var attachments []*model.FileInfo
-		attachments, err = shared.GetPostAttachments(p.Db, post)
-		if err != nil {
-			return results, err
-		}
+	var attachmentsRemovedPostIDs []string
+	allExports := make(map[string][]*ChannelExport, len(exportData.Exports))
 
-		if err = metadata.UpdateCounts(channelId, 1, len(attachments)); err != nil {
-			return results, err
-		}
+	// save a pointer to the joins for each channel, to be used later in getParticipants
+	joinsByChannel := make(map[string][]shared.JoinExport, len(exportData.Exports))
 
-		if _, ok := postAuthorsByChannel[*post.ChannelId]; !ok {
-			postAuthorsByChannel[*post.ChannelId] = make(map[string]shared.ChannelMember)
-		}
-		postAuthorsByChannel[*post.ChannelId][*post.UserId] = shared.ChannelMember{
-			UserId:   *post.UserId,
-			Username: *post.Username,
-			IsBot:    post.IsBot,
-			Email:    *post.UserEmail,
+	for _, channel := range exportData.Exports {
+		for _, post := range channel.Posts {
+			attachmentsRemoved := addToExports(rctx, allExports, channel, post)
+			attachmentsRemovedPostIDs = append(attachmentsRemovedPostIDs, attachmentsRemoved...)
 		}
+		joinsByChannel[channel.ChannelId] = channel.JoinEvents
+	}
 
-		attachmentsRemoved := addToExports(rctx, attachments, allExports, post, p.BatchStartTime, p.BatchEndTime)
-		attachmentsRemovedPostIDs = append(attachmentsRemovedPostIDs, attachmentsRemoved...)
+	tmpFile, err := os.CreateTemp("", "")
```

Comment:
Is it safe to use the default temporary directory vs something coming from config? Asking since `/tmp` on Linux may be on a separate partition with significantly less space.